From e2ba7d7ecd72c4f3aea706482b96befa30253404 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 1 Jun 2016 13:11:07 +0100
Subject: [PATCH] Don't re-typedef ?int??_t if _STDINT is defined

This is a companion patch for:
qtbase Change-Id: Ic6b850f85fd620c2ae6232e35a230f9c1f9f8afd

Change-Id: Icfd220bfcddfa644923a720ada8d0b8a95d1c0c1
---
 src/3rdparty/assimp/include/assimp/Compiler/pstdint.h | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/3rdparty/assimp/include/assimp/Compiler/pstdint.h b/src/3rdparty/assimp/include/assimp/Compiler/pstdint.h
index 5bc322f..dc5f9a2 100644
--- qt3d/src/3rdparty/assimp/include/assimp/Compiler/pstdint.h
+++ qt3d/src/3rdparty/assimp/include/assimp/Compiler/pstdint.h
@@ -303,7 +303,7 @@
 #ifndef UINT8_MAX
 # define UINT8_MAX 0xff
 #endif
-#ifndef uint8_t
+#if !defined (uint8_t) && !defined (_STDINT)
 # if (UCHAR_MAX == UINT8_MAX) || defined (S_SPLINT_S)
     typedef unsigned char uint8_t;
 #   define UINT8_C(v) ((uint8_t) v)
@@ -315,7 +315,7 @@
 #ifndef INT8_MAX
 # define INT8_MAX 0x7f
 #endif
-#ifndef INT8_MIN
+#if !defined (INT8_MIN) && !defined (_STDINT)
 # define INT8_MIN INT8_C(0x80)
 #endif
 #ifndef int8_t
@@ -330,7 +330,7 @@
 #ifndef UINT16_MAX
 # define UINT16_MAX 0xffff
 #endif
-#ifndef uint16_t
+#if !defined (uint16_t) && !defined (_STDINT)
 #if (UINT_MAX == UINT16_MAX) || defined (S_SPLINT_S)
   typedef unsigned int uint16_t;
 # ifndef PRINTF_INT16_MODIFIER
@@ -354,7 +354,7 @@
 #ifndef INT16_MIN
 # define INT16_MIN INT16_C(0x8000)
 #endif
-#ifndef int16_t
+#if !defined (int16_t) && !defined (_STDINT)
 #if (INT_MAX == INT16_MAX) || defined (S_SPLINT_S)
   typedef signed int int16_t;
 # define INT16_C(v) ((int16_t) (v))
@@ -375,7 +375,7 @@
 #ifndef UINT32_MAX
 # define UINT32_MAX (0xffffffffUL)
 #endif
-#ifndef uint32_t
+#if !defined (uint32_t) && !defined (_STDINT)
 #if (ULONG_MAX == UINT32_MAX) || defined (S_SPLINT_S)
   typedef unsigned long uint32_t;
 # define UINT32_C(v) v ## UL
@@ -405,7 +405,7 @@
 #ifndef INT32_MIN
 # define INT32_MIN INT32_C(0x80000000)
 #endif
-#ifndef int32_t
+#if !defined (int32_t) && !defined (_STDINT)
 #if (LONG_MAX == INT32_MAX) || defined (S_SPLINT_S)
   typedef signed long int32_t;
 # define INT32_C(v) v ## L
@@ -725,5 +725,9 @@ typedef uint_least32_t uint_fast32_t;
 # define SIG_ATOMIC_MAX ((((sig_atomic_t) 1) << (sizeof (sig_atomic_t)*CHAR_BIT-1)) - 1)
 #endif
 
+#if defined(_MSC_VER) && (_MSC_VER < 1600) && !defined(_STDINT)
+#  define _STDINT
+#endif
+
 #endif
 
-- 
2.8.2

